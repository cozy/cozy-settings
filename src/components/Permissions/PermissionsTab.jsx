import React, { useState } from 'react'
import Page from 'components/Page'
import { Tabs, Tab } from 'cozy-ui/transpiled/react/MuiTabs'
import AppList from './AppList'
import DataList from './DataList'
import Divider from 'cozy-ui/transpiled/react/MuiCozyTheme/Divider/index'
import { routes } from 'constants/routes'
import { useParams } from 'react-router-dom'
import withAllLocales from '../../lib/withAllLocales'

function a11yProps(index) {
  return {
    id: `simple-tab-${index}`,
    'aria-controls': `simple-tabpanel-${index}`
  }
}
const TabPanel = props => {
  const { children, value, index, ...other } = props
  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`simple-tabpanel-${index}`}
      aria-labelledby={`simple-tab-${index}`}
      {...other}
    >
      {value === index && children}
    </div>
  )
}
const PermissionsTab = ({ t }) => {
  const { page: slugOrData } = useParams()
  const [tab, setTab] = useState(slugOrData)
  const handleChange = (event, value) => setTab(value)

  return (
    <Page narrow>
      <Tabs value={tab} onChange={handleChange}>
        <Tab
          value="slug"
          label={t('Permissions.applications')}
          href={`#${routes.appList}`}
          {...a11yProps(0)}
        />
        <Tab
          value="data"
          label={t('Permissions.data')}
          href={`#${routes.dataList}`}
          {...a11yProps(1)}
        />
      </Tabs>
      <Divider className="u-mb-1" />
      <TabPanel value={tab} index="slug">
        <AppList />
      </TabPanel>
      <TabPanel value={tab} index="data">
        <DataList />
      </TabPanel>
    </Page>
  )
}

export default withAllLocales(PermissionsTab)
