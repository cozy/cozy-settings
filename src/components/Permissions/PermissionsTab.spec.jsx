import { fireEvent, render } from '@testing-library/react'
import React from 'react'

import { BreakpointsProvider } from 'cozy-ui/transpiled/react/providers/Breakpoints'

import PermissionsTab from './PermissionsTab'

jest.mock('cozy-ui/transpiled/react/providers/I18n/withLocales', () => {
  return () => Component => {
    const t = text => text
    const match = { params: { page: 'slug' } }
    return () => <Component match={match} t={t} />
  }
})
const mockNavigate = jest.fn()
jest.mock('react-router-dom', () => {
  return {
    useParams: () => ({ page: 'slug' }),
    useNavigate: () => mockNavigate
  }
})

jest.mock('cozy-client')
jest.mock('@/components/PageTitle', () => {
  return ({ children }) => <div data-testid="pageTitle">{children}</div>
})
jest.mock(
  'cozy-ui/transpiled/react/Tabs',
  () =>
    ({ value, onChange, children }) =>
      (
        <button
          value={value}
          onClick={event => onChange(event, 'data')}
          data-testid="tabs"
        >
          {children}
        </button>
      )
)

jest.mock(
  'cozy-ui/transpiled/react/Tab',
  () =>
    ({ value, label, href, id, 'aria-controls': ariaControls }) =>
      (
        <div
          data-testid="tab"
          value={value}
          label={label}
          href={href}
          id={id}
          aria-controls={ariaControls}
        ></div>
      )
)

jest.mock('./AppList/AppList', () => {
  return () => <div data-testid="AppList"></div>
})

jest.mock('./DataList/DataList', () => {
  return () => <div data-testid="DataList"></div>
})

describe('PermissionsTab', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <BreakpointsProvider>
        <PermissionsTab />
      </BreakpointsProvider>
    )
    expect(container).toMatchSnapshot()
  })
  it('should display AppList when Application Tab is selected', () => {
    const { queryByTestId } = render(
      <BreakpointsProvider>
        <PermissionsTab />
      </BreakpointsProvider>
    )
    expect(queryByTestId('AppList')).toBeTruthy()
  })

  it('should display DataList when Data Tab is selected', () => {
    const { getByTestId } = render(
      <BreakpointsProvider>
        <PermissionsTab />
      </BreakpointsProvider>
    )
    const tabs = getByTestId('tabs')
    fireEvent.click(tabs, { event: 'event' })
    expect(mockNavigate).toHaveBeenCalledWith('/permissions/data')
  })
})
