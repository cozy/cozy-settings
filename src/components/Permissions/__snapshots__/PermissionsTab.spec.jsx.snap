// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PermissionsTab should match snapshot 1`] = `
<div>
  <div
    class="u-mh-2 u-pb-3 u-mv-2"
  >
    <div>
      <div
        class="u-flex"
      >
        <div
          data-testid="pageTitle"
        >
          Permissions.permissions
        </div>
        <button
          data-testid="tabs"
          value="slug"
        >
          <div
            aria-controls="simple-tabpanel-0"
            data-testid="tab"
            href="#/permissions/slug"
            id="simple-tab-0"
            label="Permissions.applications"
            value="slug"
          />
          <div
            aria-controls="simple-tabpanel-1"
            data-testid="tab"
            href="#/permissions/data"
            id="simple-tab-1"
            label="Permissions.data"
            value="data"
          />
        </button>
      </div>
      <div
        aria-labelledby="simple-tab-slug"
        id="simple-tabpanel-slug"
        role="tabpanel"
      >
        <div
          data-testid="AppList"
        />
      </div>
      <div
        aria-labelledby="simple-tab-data"
        hidden=""
        id="simple-tabpanel-data"
        role="tabpanel"
      />
    </div>
  </div>
</div>
`;
